template-for-repository/negative_tests/nan_check/nan_check_harness.c

```diff
@@ -0,0 +1,12 @@
+/**
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+
+/**
+ * A negative test for --nan-check flag
+ */
+void nan_check_harness() {
+    float nan;
+    nan = nan / nan;
```

Comment:
Could this also trigger the div by 0 error?

Reply:
Yes, this would trigger both. I wasn't sure how to _only_ trigger a `nan` (in this case it would be `0/0`).